Scripts/AssemblyInjections/Unity.Entities/EntityCommandBufferParallelWriterInternal.cs

```diff
@@ -0,0 +1,21 @@
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Helper class to access the internals of a <see cref="EntityCommandBuffer.ParallelWriter"/>
+    /// </summary>
+    public static class EntityCommandBufferParallelWriterInternal
+    {
+        /// <summary>
+        /// Sets the Thread Index on a <see cref="EntityCommandBuffer.ParallelWriter"/> so that commands are written
+        /// to the correct Thread Chain
+        /// </summary>
+        /// <param name="ecbParallelWriter">The <see cref="EntityCommandBuffer.ParallelWriter"/></param>
+        /// <param name="threadIndex">The thread index to set</param>
+        public static void SetThreadIndex(ref this EntityCommandBuffer.ParallelWriter ecbParallelWriter, int threadIndex)
```

Comment:
Can we add a `<remark>` that mentions this shouldn't usually be required and describe a situation where it's necessary?

Reply:
Yes done.